lib/routes/receive_payment/lnurl/lnurl_withdraw_page.dart

```diff
@@ -44,36 +42,75 @@ class LnUrlWithdrawPageState extends State<LnUrlWithdrawPage> {
   final _amountFocusNode = FocusNode();
   var _doneAction = KeyboardDoneAction();
 
-  late LightningPaymentLimitsResponse _lightningLimits;
-
-  PrepareReceiveResponse? prepareResponse;
-  Future<ReceivePaymentResponse>? receivePaymentResponse;
-
-  bool isBelowPaymentLimit = false;
+  bool _isFixedAmount = false;
+  bool _loading = true;
+  String? _errorMessage;
+  LightningPaymentLimitsResponse? _lightningLimits;
 
   @override
   void initState() {
     super.initState();
     _doneAction = KeyboardDoneAction(focusNodes: [_amountFocusNode]);
 
-    WidgetsBinding.instance.addPostFrameCallback((_) {
+    WidgetsBinding.instance.addPostFrameCallback((_) async {
+      await _fetchLightningLimits();
       _initializePageData();
     });
   }
 
+  Future<void> _fetchLightningLimits() async {
+    final paymentLimitsCubit = context.read<PaymentLimitsCubit>();
+    try {
+      final response = await paymentLimitsCubit.fetchLightningLimits();
+      _handleLightningPaymentLimitsResponse(response);
+    } catch (error) {
+      setState(() {
+        _errorMessage = error.toString();
+        _loading = false;
+      });
+    }
+  }
+
+  void _handleLightningPaymentLimitsResponse(LightningPaymentLimitsResponse response) {
+    final minWithdrawableSat = widget.requestData.minWithdrawable.toInt() ~/ 1000;
+    final maxWithdrawableSat = widget.requestData.maxWithdrawable.toInt() ~/ 1000;
+    final effectiveMinSat = max(response.receive.minSat.toInt(), minWithdrawableSat);
+    final effectiveMaxSat = min(response.receive.maxSat.toInt(), maxWithdrawableSat);
+    if (effectiveMaxSat < effectiveMinSat) {
+      final texts = context.texts();
+      final currencyCubit = context.read<CurrencyCubit>();
+      final currencyState = currencyCubit.state;
+
+      final isFixedAmountWithinLimits = _isFixedAmount && (effectiveMinSat == effectiveMaxSat);
```

Comment:
I think `_isFixedAmount` has not been set yet as `_initializePageData()` is called after